app/models/mixins/per_ems_worker_mixin.rb

```diff
@@ -20,22 +20,23 @@ def ems_class
     end
 
     def all_ems_in_zone
-      ems_class.where(:zone_id => MiqServer.my_server.zone.id).to_a
+      ems_class.where(:zone_id => MiqServer.my_server.zone.id)
     end
 
     def all_valid_ems_in_zone
-      all_ems_in_zone.select {|e| e.enabled && e.authentication_status_ok?}
+      all_ems_in_zone.select { |e| e.enabled && e.authentication_status_ok? }
```

Comment:
Would we want to change this back to the `where` version originally introduced in https://github.com/ManageIQ/manageiq/pull/14675/files#diff-2e5c585febf1bb6ea7c1e0205e378906R21 ?

Also I see that this is just a rubocop fix, but since it's in the patch it would probably be a good enhancement.

Related: I know it's a small change in this case, but it would be much better if the rubocop fixes were a separate commit from the actual change. It makes it easier to review and also easier to reference the important part of the patch in the future.

Reply:
Also it sounds like changing this back to `where` was the original goal of https://github.com/ManageIQ/manageiq-providers-azure/pull/310, right?

Reply:
@kbrock thoughts?

Reply:
using a `where()` is the reason for getting rid of the `to_a` in the first place.

If we mostly used enabled when fetching ems, then we'd use that query cached value locally. It would perform better. 
But I feel like we aren't consistent here. So if we use both, and each one has a different query, then we will be double fetching the ems records.

my gut says it is probably cheaper to fetch all records.

Assumptions:
- most of the ems are enabled.
- there are only a dozen ems
Assum